src/moscot/problems/_utils.py

```diff
@@ -85,6 +83,26 @@ def handle_joint_attr(
         }
         return xy, kwargs
     if isinstance(joint_attr, Mapping):
+        joint_attr = dict(joint_attr)
+        if "attr" in joint_attr and joint_attr["attr"] == "X":
+            return {"x_attr": "X", "y_attr": "X"}, kwargs
+        if "attr" in joint_attr and joint_attr["attr"] == "obsm":
+            if "key" not in joint_attr:
```

Comment:
Just a small note, no need to change anything: slightly more "Pythonic" way would be to
ask for forgiveness rather than a permission as in:
```python
try:
    xy = ...
except KeyError:
    raise KeyError("`key` must be provided when ...")